packages/flutter/lib/src/animation/curves.dart

```diff
@@ -267,6 +273,526 @@ class Cubic extends Curve {
   }
 }
 
+/// Abstract class that defines an API for evaluating 2D parametric curves.
```

Comment:
This should describe how this is different from a regular "Curve"

Reply:
Isn't Curve also a 2D curve?

Reply:
The only difference seems to be that Curve maps t=1.0 and t=0.0 to 1.0 and 0.0, respectively, while this curve doesn't have this requirement?

Reply:
It also seems somewhat confusing that a Curve2D is not a Curve (or vice versa).

Reply:
The naming is confusing.

Reply:
I can see how it can be confusing, but no, it's not a 2D curve in the sense that the output of the parametric curve is a single double value, while the output of the `Curve2D` class is an `Offset`.  I.e. `Curve` is mapping f(t) -> x, where `Curve2D` is mapping `f(t) -> (x, y)`.